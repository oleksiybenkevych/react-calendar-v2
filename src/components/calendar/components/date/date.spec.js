import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";

import Date from "./date";
import moment from "moment";

const mockStore = configureStore([]);
describe("Date", () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      range: { start: moment(), end: moment() },
      hoverEndDate: moment(),
      firstClick: false
    });
    store.dispatch = jest.fn();
  });

  it("should render with given state from Redux store", () => {
    const DateComponent = renderer.create(
      <Provider store={store}>
        <Date />
      </Provider>
    );
    let component = DateComponent.toJSON();
    expect(component).toMatchSnapshot();
  });
});
