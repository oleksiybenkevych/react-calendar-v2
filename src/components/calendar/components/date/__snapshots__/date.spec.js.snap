// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date should render with given state from Redux store 1`] = `
<td
  className="empty-day"
/>
`;
